core/versions_2.0.md

```diff
@@ -0,0 +1,71 @@
+# Workflow versioning 2.0
+
+This proposal introduces two key changes to what is called `getVersion` in the existing SDKs, but will likely be renamed in subsequent ones (more on naming later). The key changes are:
+
+* Introduce a different API for versioning that does not involve a version range, but instead simply returns a boolean indicating if a named change has happened or not. Ex: `hasChange("change-id")`. TBD if we would eliminate the range-based API, or leave it for advanced users.
+* It will no longer be valid to replay a workflow history containing a version marker against code which does not have a corresponding call to the version API.
+
+## Boolean-only API and Renaming
+The existing `getVersion` API is confusing to users. What's a default version? What should the maximum be? Why is it always returned by workflows that have hit that line for the first time?
+
+If it's the case that very few users ever actually need version ranges, which seems likely, then a much simpler API can be presented. In fact, it's still possible to conceptually represent version ranges with it as well, if needed.
+
+Instead of `get_version("change-id", minver, maxver) -> number`, consider: `changed("change-id") -> bool`
+
+If the workflow code is executing for the first time, it will write a marker into history meaning "change change-id happened". If it is replaying history, it will verify such a marker exists and return true. If the marker does not exist, it will return false. Versions of the code without the call will fail (more on that in other section).
+
+Version ranges can still be simulated if needed like:
+```rust
+if changed("change-ver-1") {
```

Comment:
I think you want to swap "change-ver-1" and "change-ver-2" here, otherwise the 2nd clause will never apply

Reply:
Ah, yeah that's a good point thanks